Add database

Elite-tea opened the pull request:
feat: Добавлена работа с БД.

Elite-tea committed:
feat:Добавлена схема для будущей реализации БД.
feat:Добавлены поля genre и mpa для возможности указания жанров и рейтинга фильма.

Elite-tea committed:
feat:Добавлена реализация хранения данных в БД
feat:Добавлены методы для получения данных жанров
feat:Добавлены методы для получения данных рейтинга

Elite-tea committed:
feat:Удалены лишние методы

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmDbService.java:
```diff
@@ -0,0 +1,164 @@
+    private final LikeDao likeDao;
+
+    @Autowired
+    public FilmDbService(@Qualifier("FilmDbStorage") FilmDbStorage filmStorage, @Qualifier("UserDbStorage") UserDbStorage userStorage, GenreDao genreDao, MpaDao mpaDao, LikeDao likeDao) {
```
Когда в сигнатуре метода много аргументов, рекомендуется переносить каждый аргумент на отдельную строку для лучшей читаемости.

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmDbService.java:
```diff
@@ -0,0 +1,164 @@
+     */
+
+    public List<Film> getPopularFilm(int topNumber) {
+        return getFilm().stream().sorted(this::compare).limit(topNumber).collect(Collectors.toList());
```
Стрим легче читать, когда каждая операция на отдельной строке

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -1,122 +1,124 @@
+//@Service
+//@RequiredArgsConstructor
+//@Slf4j
+//public class FilmService {
```
Не следует держать в репозитории закоментированные куски кода, а тем более целые классы

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/GenreDbService.java:
```diff
@@ -0,0 +1,27 @@
+
+    public Genre getGenreById(Integer id) {
+        if (id == null) {
+            throw new NotFoundException("Жанра не существует");
```
Я как обычно рекомендую добавлять служебную информацию в сообщение: Жанра с id 111 не существует. Так легче в будущем разбирать ошибки

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/UserDbService.java:
```diff
@@ -0,0 +1,109 @@
+    private final FriendDao friendDao;
+
+    @Autowired
+    public UserDbService(@Qualifier("UserDbStorage") UserDbStorage userStorage, GenreDao genreDao, MpaDao mpaDao, LikeDao likeDao, FriendDao friendDao) {
```
Здесь тоже было бы здорово разбить аргументы на несколько строк

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/UserDbService.java:
```diff
@@ -0,0 +1,109 @@
+
+        log.info("Запрошены общие друзья у пользователя с id {} и {}", userId, idFriend);
+
+        return friendFriends.stream().filter(userFriends::contains).filter(friendFriends::contains).collect(Collectors.toList());
```
Здесь тоже бы разбить стрим на несколько строк

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/UserDbService.java:
```diff
@@ -0,0 +1,109 @@
+    public List<User> getFriends(Long id) {
+        if (!userStorage.getByIdUser(id).getEmail().isEmpty()) {
+            log.info("Запрошены друзья у пользователя с id {}", id);
+            return friendDao.getFriend(id).stream().mapToLong(Long::valueOf).mapToObj(userStorage::getByIdUser).collect(Collectors.toList());
```
И здесь тоже на несколько строк

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/UserDbService.java:
```diff
@@ -0,0 +1,109 @@
+        if (!userStorage.getByIdUser(id).getEmail().isEmpty()) {
+            log.info("Запрошены друзья у пользователя с id {}", id);
+            return friendDao.getFriend(id).stream().mapToLong(Long::valueOf).mapToObj(userStorage::getByIdUser).collect(Collectors.toList());
+        } else {
```
Некоторые разработчики предпочитают менять условие для лучшей читаемости кода (но это тоже спорно). Но все же:

```
if (userStorage.getByIdUser(id).getEmail().isEmpty()) {
   throw new Exception();
}
return friendDao.getFriend(id)...;
```

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -1,116 +1,116 @@
-    }
-}
\ No newline at end of file
+//package ru.yandex.practicum.filmorate.service;
```
Ну вот вообще некрасиво так комментировать код

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/storage/dao/FriendDaoImpl.java:
```diff
@@ -0,0 +1,55 @@
+
+@AllArgsConstructor
+@Component
+public class FriendDaoImpl implements FriendDao {
```
В этом классе очень много длинных строк. Попытайтесь их разбить для лучшего прочтения кода

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/storage/dao/UserDaoImpl.java:
```diff
@@ -0,0 +1,42 @@
+
+@AllArgsConstructor
+@Component
+public class UserDaoImpl implements UserStorage {
```
Здесь тоже бы разбить длинные строки

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java:
```diff
@@ -0,0 +1,62 @@
+@Slf4j
+@Component("FilmDbStorage")
+@RequiredArgsConstructor
+public class FilmDbStorage implements FilmStorage {
```
И здесь разбить на несколько строк длинные строки

kriwsha commented on src/test/java/ru/yandex/practicum/filmorate/FilmTest.java:
```diff
@@ -1,72 +1,72 @@
-
-}
\ No newline at end of file
+//package ru.yandex.practicum.filmorate;
```
Не надо так делать

Elite-tea committed:
feat:Добавлена документация на некоторые методы и классы
style: Исправлены длинные строки кода для улучшения чтения
style: Улучшено визуальное восприятие кода
delete: Удалены неиспользуемые классы

Elite-tea committed:
fix: Удалены не используемые импорты

Elite-tea committed:
fix: Удалены еще не используемые импорты

Elite-tea committed:
delete: Удалены неиспользуемые методы
delete: Удалены неиспользуемые классы
feat: Дописана вся документация по всем методам
refactor: Dao разложены по соответствующим папкам
